docs/events_handling.md

```diff
@@ -0,0 +1,102 @@
+
+# Events handling
+
+## Subscribed events
+
+Precaution needs to subscribe to a number of events
+to satisfy different use cases of our users.
+
+The events are separated into two groups `pull requests
+events` and `check suites/runs events`.
+
+### Pull requests events
+
+Official GitHub documentation: [PullRequest event](https://developer.github.com/v3/activity/events/types/#pullrequestevent)
+
+* <strong>pull_request.opened</strong>
+
+*occurs - when a new pull request is created*
+
+*triggers - start of the Precaution scan*
+
+* <strong>pull_request.reopened</strong>
+
+*occurs - when a pull request is reopened from a closed state*
+
+*triggers - start of a new Precaution scan*
+
+* <strong>pull_request.synchronize</strong>
+
+*occurs - when you add a commit to already opened pull request*
+
+*triggers - start of a new Precaution scan*
+
+*for more information look at pull request number [#113](https://github.com/vmware/precaution/pull/113)*
+
+
+### Check run and check suite events
+
+Official GitHub documentation:
+- [CheckRunEvent](https://developer.github.com/v3/activity/events/types/#checkrunevent)
+- [CheckSuiteEvent](https://developer.github.com/v3/activity/events/types/#checksuiteevent)
+
+A user had already opened a pull request and he is at the pull request page.
+He navigates to the "Checks" tab.
+If the Precaution check had failed he will see "Re-run all checks", "Re-run" and
+"Re-run failed checks" buttons.
+
+* <strong>check_run.rerequested</strong>
+
+*occurs - when you click the "Re-run" button*
+
+*triggers - start of a new Precaution scan*
+
+* <strong>check_suite.rerequested</strong>
+
+*occurs - when you click "Re-run all checks"
+or "Re-run failed checks" buttons*
+
+*triggers - start of a new Precaution scan*
+
+## Events workflows
+
+This paragraph doesn't aim to explain in details how Precaution works.
+If you want to read more details how Precaution works you can look at projects [Architecture](architecture.md).
```

Comment:
This reads a bit awkwardly as there's repeated phrasing, I suggest we change the second sentence to something like

```suggestion
If you want to understand more details of Precaution's design see the [Architecture](architecture.md) documentation.
```